frontend/src/components/dashboard/ProfileBanners.tsx

```diff
@@ -22,12 +22,14 @@ import { trackPurchaseStart } from "../../functions/trackPurchase";
 import { renderDate } from "../../functions/renderDate";
 import { SubdomainPicker } from "./SubdomainPicker";
 import { useMinViewportWidth } from "../../hooks/mediaQuery";
+import { AliasData } from "../../hooks/api/aliases";
 
 export type Props = {
   profile: ProfileData;
   user: UserData;
   onCreateSubdomain: (chosenSubdomain: string) => Promise<void>;
   runtimeData?: RuntimeData;
+  aliases: AliasData[];
```

Comment:
@Vinnl I tried adding `props.alias.length`  as a condition for showing the Premium banner under ProfileBanners.tsx, but I am getting the `can't access property "length", props.aliases is undefined` error - I've omitted it for now but was wondering why you think this error is being thrown, since the length property seems to work in other components (see `AliasGenerator.tsx` and `AliasList.tsx`)

Reply:
If you look at [the failed check](https://app.circleci.com/pipelines/github/mozilla/fx-private-relay/4933/workflows/95c2fe6c-b954-41b3-a431-b9400f263571/jobs/10852) linked below, you can see the following error:

```
./src/pages/accounts/profile.page.tsx:303:8
Type error: Property 'aliases' is missing in type '{ profile: ProfileData; user: UserData; onCreateSubdomain: (customSubdomain: string) => Promise<void>; runtimeData: RuntimeData; }' but required in type 'Props'.

  301 |   const banners = (
  302 |     <section className={styles["banners-wrapper"]}>
> 303 |       <ProfileBanners
      |        ^
  304 |         profile={profile}
  305 |         user={user}
  306 |         onCreateSubdomain={setCustomSubdomain}
```

So while you indicated in the `Props` type of `<ProfileBanners>` that you want it to be passed a list of aliases, no such list actually gets passed in yet where the component is included (i.e. in `profile.page.tsx`). Luckily, that component does already have the list of the user's aliases (in the `allAliases` variable), so you have it give that to the `<ProfileBanners>` component.

Also, note that you named it `aliases`, so you'll need `props.aliases.length` rather than `props.alias.length` :)

Reply:
Passing aliases to the `<ProfileBanners>` worked, thank you Vincent! Does this mean that it'll throw an error if I don't utilize all of the props when passing a component to pages where its being injected?

Reply:
Basically you can define how your component should be called (i.e. when it will throw an error) via the `Props` type. In this case, you defined it as adding an `aliases` attribute ("prop") that should be passed an array of `AliasData` objects:

https://github.com/mozilla/fx-private-relay/blob/f94960c613f61ff74243bf84514f02f03e092fb5/frontend/src/components/dashboard/ProfileBanners.tsx#L32

You could also have defined it as accepting either that array, or `undefined`, effectively making it optional:

```ts
  aliases: AliasData[] | undefined;
```

That's actually such a common use case that there's a shorthand for it:

```ts
  aliases?: AliasData[];
```

(i.e. the `?` marks it as optional)